crypto/ec_extra/ec_asn1.c

```diff
@@ -614,3 +614,44 @@ BIGNUM *EC_POINT_point2bn(const EC_GROUP *group, const EC_POINT *point,
 
   return ret;
 }
+
+EC_POINT *EC_POINT_bn2point(const EC_GROUP *group,
+                            const BIGNUM *bn, EC_POINT *point, BN_CTX *ctx) {
+  EC_POINT *ret = NULL;
+
+  // Allocate buffer and length.
+  size_t buf_len = BN_num_bytes(bn);
+  if (buf_len == 0) {
+    buf_len = 1;
+  }
+  uint8_t *buf = OPENSSL_malloc(buf_len);
+  if (buf == NULL) {
+    return NULL;
+  }
+
+  if (BN_bn2bin_padded(buf, buf_len,bn) < 0) {
+    goto end;
+  }
+
+  // Allocate new |EC_POINT| if |point| is NULL. Otherwise, use |point|.
+  if (point == NULL) {
+    ret = EC_POINT_new(group);
+    if (ret == NULL) {
+      goto end;
+    }
+  } else {
+    ret = point;
+  }
+
+  if (!EC_POINT_oct2point(group, ret, buf, buf_len, ctx)) {
+    if (ret != point) {
+      // Free the newly allocated |EC_POINT| on failure.
```

Comment:
I'm not sure I understand the logic around `ret`/`point`. can we implement that in a simpler way?

Reply:
The API design around this wasn't the best. The user can provide an `EC_POINT` with `point` and use that to save the intended value. If it's not provided, OpenSSL/AWS-LC allocates a value for you. Basically we want to be careful not to free the `EC_POINT` if the user has provided one, but we should be freeing the one we allocated.

I rearranged a code a bit to make more readable, do lmk if it's still a bit confusing